mlem/core/dataset_type.py

```diff
@@ -48,6 +51,10 @@ def check_type(obj, exp_type, exc_type):
                 f"given dataset is of type: {type(obj)}, expected: {exp_type}"
             )
 
+    @abstractmethod
+    def combine(self, batched_data: List[List[T]]) -> List[T]:
```

Comment:
Actually, come to think of it, I used `(List[List[T]) -> List[T])` because T is a `TypeVar`, type checks via mypy will be incorrect if we do `(List[T]) -> T` instead.

Reply:
I mean, np.concatenate accepts a sequence of array-like and you should pass it a `List[np.array]`. And it returns siingle `np.array`. So correct signature is `List[T] -> T` where `T` is `np.array` or `pd.DataFrame` of something like this